core/src/saros/negotiation/ProjectNegotiationData.java

```diff
@@ -54,4 +74,13 @@ public String getProjectID() {
   public boolean isPartial() {
     return partial;
   }
+
+  /**
+   * Returns an unmodifiable view of the map of additional project options.
```

Comment:
Or null if a Saros client with a former version sent this packet. So question is: do you want to make the protocol incompatible ?

Reply:
Good point. I haven't thought of that. Can't think of a solution that is not breaking compatibility off the top of my head.

For Saros/I, this is much of a problem in my opinion. We don't have a lot of users and, more importantly, no version compatibility table, so every new version is seen as incompatible by default (but this should be fixed sooner than later; together with adjusting the existing logic to differentiate IDE implementations). Furthermore, the new module creation logic is dependent on the passed information, so backwards compatibility is not really an option.

But the issue would be Saros/E. I am not sure what would happen if an outdated version is setting up the session. Would the value really just be `null`? In that case, it wouldn't be an issue *yet* as Saros/E is not using the logic. On the other hand, if the usage is going to be added in the foreseeable future, it might be better to make a clean cut and break compatibility now while the plugin is still a bit in "turmoil" (i.e. a lot of stuff is happening/changing internally).

What are your opinions on the matter?

Reply:
I do not see an issue to return an empty map if none was given. You should verify the content anyway.